app/lib/frontend/handlers.dart

```diff
@@ -233,14 +233,17 @@ Future<shelf.Response> staticsHandler(shelf.Request request) async {
         !staticFile.lastModified.isAfter(ifModifiedSince)) {
       return new shelf.Response.notModified();
     }
-    // TODO: handle ETag header
+    final ifNoneMatch = request.headers[HttpHeaders.IF_NONE_MATCH];
+    if (ifNoneMatch != null && ifNoneMatch == staticFile.etag) {
+      return new shelf.Response.notModified();
+    }
     return new shelf.Response.ok(
       staticFile.bytes,
       headers: {
         HttpHeaders.CONTENT_TYPE: staticFile.contentType,
         HttpHeaders.CONTENT_LENGTH: staticFile.bytes.length.toString(),
         HttpHeaders.LAST_MODIFIED: formatHttpDate(staticFile.lastModified),
```

Comment:
Remove this one, then. Times may change based on deploys...but etags are valid as long as the content stays the same